client/app/components/toolbox_actions.coffee

```diff
@@ -1,166 +1,144 @@
         direction            : React.PropTypes.string.isRequired
-        displayConversations : React.PropTypes.bool.isRequired
+        # one of conversation / message
+        mode                 : React.PropTypes.string.isRequired
+        # is the message or all messages flagged
         isFlagged            : React.PropTypes.bool
+        # is the message or all messages seen
         isSeen               : React.PropTypes.bool
+        # mailboxes this message can be moved to
         mailboxes            : React.PropTypes.object.isRequired
-        message              : React.PropTypes.object
+        # id of the message we are working on (empty for conversation)
         messageID            : React.PropTypes.string
+        # handlers for action
         onConversationDelete : React.PropTypes.func.isRequired
         onConversationMark   : React.PropTypes.func.isRequired
         onConversationMove   : React.PropTypes.func.isRequired
         onHeaders            : React.PropTypes.func
-        onMark               : React.PropTypes.func.isRequired
+        onMark               : React.PropTypes.func
 
 
     shouldComponentUpdate: (nextProps, nextState) ->
         return not(_.isEqual(nextState, @state)) or
                not(_.isEqual(nextProps, @props))
 
-
     render: ->
-        direction = if @props.direction is 'right' then 'right' else 'left'
-        div className: 'menu-action btn-group btn-group-sm',
-            button
-                className: 'btn btn-default dropdown-toggle fa fa-cog'
-                type: 'button'
-                'data-toggle': 'dropdown'
-                ' '
-                    span className: 'caret'
-            ul
-                className: "dropdown-menu dropdown-menu-#{direction}"
-                role: 'menu',
-                    # in conversation mode, only shows actions on conversation
-                    if not @props.displayConversations
-                        @renderMarkActions()
-                    if not @props.displayConversations
-                        MenuDivider()
-                    @renderRawActions()...
-                    if @props.inConversation
-                        @renderConversationActions()
-                    if @props.inConversation
-                        MenuDivider key: 'divider'
-                    if @props.inConversation
-                        MenuHeader key: 'header-move',
-                            t 'mail action conversation move'
-                    if @props.inConversation
-                        @renderMailboxes()
-
-
-    renderMarkActions: ->
-        items = [
-            MenuHeader key: 'header-mark', t 'mail action mark'
-
-            if not @props.isSeen? or not @props.isSeen
+        message = @props.mode is 'message'
+        conversation = @props.mode is 'conversation'
+        Menu
+            icon: 'fa-cog'
+            direction: if @props.direction is 'right' then 'right' else 'left'
+        ,
```

Comment:
React components takes an attributes object as first param and then the children as nullable varargs, I needed the coma to make coffee compile this part.